types.cc

```diff
@@ -947,6 +948,79 @@ cql3::cql3_type abstract_type::as_cql3_type() const {
     return cql3::cql3_type(shared_from_this());
 }
 
+namespace {
+struct panic_collection_visitor {
```

Comment:
this whole structure looks redundant; you can provide a generic operator for `const abstract_type& t`, which would serve as a default handler. Then, you don't need to handle all the types explicitly.

Reply:
I tried with `const auto&` but it didn't compile. With `abstract_type` it does.